lib/loader.js

```diff
@@ -35,6 +30,13 @@ function sassLoader(content) {
         throw new Error("Synchronous compilation is not supported anymore. See https://github.com/webpack-contrib/sass-loader/issues/333");
     }
 
+    const compilerError = asyncJobQueue.error;
+
+    if (compilerError) {
```

Comment:
No need to do that check. If the error was just thrown, webpack would report the loader error accordingly.